### ISSUE TITLE
Remote wallet management for lotus-miner 

### ISSUE THREAD

**Issue by jennijuju** [filecoin-project/lotus#7670]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [X] Other

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

to provide more flexibility to access miner box for operations without concerning about leaking the wallet keys.

### Describe the solution you'd like

- wallet being hosted on one machine
- lotus miner has api access to the wallet node via a token 
- one can define the perm of the api token, i.e: a set of API that token is granted permission

**Comment by stuberman** [comment]
We understand that the private key is passed in the clear with API calls, please encrypt the data transfers. For a remote wallet this is extremely important.

**Comment by jennijuju** [comment]
i dont think we should ever pass pk in api param - thats sounds like a terrible idea and def should avoid.

**Comment by travisperson** [comment]
I messed around bit with this a while back for a few nights.

https://github.com/travisperson/secure-wallet
